Geometry/MTDGeometryBuilder/src/MTDGeomUtil.cc

```diff
@@ -0,0 +1,206 @@
+  const ProxyMTDTopology& topoproxy = static_cast<const ProxyMTDTopology&>(thedet->topology());
+  const RectangularMTDTopology& topo = static_cast<const RectangularMTDTopology&>(topoproxy.specificTopology());
+  const Local3DPoint local_point(topo.localX(row), topo.localY(column), 0.);
+  return topo.pixel(local_point);
+}
+
+// returns row and column as a pair (row, col)
+std::pair<uint8_t, uint8_t> MTDGeomUtil::getPixelInModule(const DetId& id, const LocalPoint& local_point) const {
+  if (isETL(id)) {
+    ETLDetId detId(id);
+    DetId geoId = detId.geographicalId();
+    const MTDGeomDet* thedet = getGeometry()->idToDet(geoId);
+    if (thedet == nullptr)
+      throw cms::Exception("mtd::MTDGeomUtil") << "GeographicalID: " << std::hex << geoId.rawId() << " ("
+                                               << detId.rawId() << ") is invalid!" << std::dec << std::endl;
+    const ProxyMTDTopology& topoproxy = static_cast<const ProxyMTDTopology&>(thedet->topology());
+    const RectangularMTDTopology& topo = static_cast<const RectangularMTDTopology&>(topoproxy.specificTopology());
+    const auto& thepixel = topo.pixel(local_point);
+    uint8_t row(thepixel.first), col(thepixel.second);
+    return std::pair<uint8_t, uint8_t>(row, col);
+  } else {
+    BTLDetId detId(id);
+    DetId geoId = detId.geographicalId(MTDTopologyMode::crysLayoutFromTopoMode(getTopology()->getMTDTopologyMode()));
+    const MTDGeomDet* thedet = getGeometry()->idToDet(geoId);
+    if (thedet == nullptr)
+      throw cms::Exception("mtd::MTDGeomUtil") << "GeographicalID: " << std::hex << geoId.rawId() << " ("
+                                               << detId.rawId() << ") is invalid!" << std::dec << std::endl;
+    const ProxyMTDTopology& topoproxy = static_cast<const ProxyMTDTopology&>(thedet->topology());
+    const RectangularMTDTopology& topo = static_cast<const RectangularMTDTopology&>(topoproxy.specificTopology());
+    auto topo_point = topo.pixelToModuleLocalPoint(local_point, detId.row(topo.nrows()), detId.column(topo.nrows()));
+    const auto& thepixel = topo.pixel(topo_point);
+    uint8_t row(thepixel.first), col(thepixel.second);
+    return std::pair<uint8_t, uint8_t>(row, col);
+  }
+}
+
+int MTDGeomUtil::getCrystalInModule(const DetId& id) const {
+  BTLDetId hid(id);
+  return hid.crystal();
+}
+
+float MTDGeomUtil::getEta(const GlobalPoint& position, const float& vertex_z) const {
+  GlobalPoint corrected_position = GlobalPoint(position.x(), position.y(), position.z() - vertex_z);
+  return corrected_position.eta();
+}
+
+float MTDGeomUtil::getEta(const DetId& id, const LocalPoint& local_point, const float& vertex_z) const {
+  GlobalPoint position = getPosition(id, local_point);
+  float eta = getEta(position, vertex_z);
+  return eta;
+}
+
+float MTDGeomUtil::getPhi(const GlobalPoint& position) const {
+  float phi = atan2(position.y(), position.x());
+  return phi;
```

Comment:
@AuroraPerego although academic, perhaps better to add a protection as suggested in https://twiki.cern.ch/twiki/bin/view/CMS/CMSConventions#The_azimuthal_angle_phi_in_CMS for the 0,0 case